### ISSUE TITLE
"Failed to read the 'cssRules' property from 'CSSStyleSheet'"

### ISSUE THREAD

**Issue by jakubkloc** [broofa/checkcss#7]
I install package and try to use it in my index.jsx 
```
import React from "react";
import ReactDOM from "react-dom/client";
import App from "./App";

import { CheckCSS } from 'checkcss';

// Create CheckCSS instance
const checkcss = new CheckCSS();
checkcss.scan().watch();

ReactDOM.createRoot(document.querySelector("#root")).render(<App />);
```
and when I run npm run dev I have errors in conosole:
![image](https://user-images.githubusercontent.com/99250816/226494020-da0bf716-f63e-4186-9311-fa697da840f4.png)

**Comment by broofa** [comment]
Are you importing one or more stylesheets from a host other than `localhost:5173`?  It may be you're running into a CORS or CSP issue.

It would be very helpful if you could provide a "minimal, reproducible, verifiable" example that illustrates the problem.  Maybe start by [forking this CodePen template](https://codepen.io/broofa/pen/RwYYOdQ?editors=1112)?

**Comment by broofa** [comment]
Diving a bit deeper, my best guess is you have a stylesheet `<link>` that points to an external domain.  If that's the case, you might try adding the [`crossorigin`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/crossorigin) attribute to see if that fixes the problem.  (Requires the server support CORS, however.)

See also https://stackoverflow.com/questions/3211536/accessing-cross-domain-style-sheet-with-cssrules

Closing this issue out for the time being as it's not actionable without more information about what's causing this problem.  If you can provide a test case for reproducing the problem, I'll reopen and investigate further.

**Comment by shshaw** [comment]
Suggesting a reopening.

This came up recently due to a third party script injecting a third party stylesheet _without_ a `crossorigin` attribute. I can't exactly modify the script or that `<link />` it injects, so some way to indicate to CheckCSS that it should be ignored would be helpful.

As I don't think there'd be a good way to detect and ignore external files, my recommendation would be to add a `try...catch` around accessing stylesheets to safely report the error, rather than grinding everything to a halt.

**Comment by broofa** [comment]
@shshaw:

**Comment by broofa** [comment]
@shshaw : It's simple enough to throw a try-catch around that logic, but this creates a problem where any references to styles defined by the stylesheet will be logged as undefined.  The only workaround I can think of for that would be to detect and exclude those classes in `onClassnameDetected`.

**Comment by broofa** [comment]
@Kubot11 @shshaw: Should be fixed in #8.  Published as `checkcss@2.0.6`.   As noted previously you may need to whitelist classes defined by external stylesheets using the `onClassnameDetected` callback.
